src/query-hoc.tsx

```diff
@@ -51,7 +51,7 @@ export function query<
       render() {
         let props = this.props;
         const shouldSkip = mapPropsToSkip(props);
-        const opts = shouldSkip ? Object.create(null) : mapPropsToOptions(props);
+        const opts = shouldSkip ? Object.create(null) : { ...mapPropsToOptions(props) };
```

Comment:
`options` can be passed into the `graphql` hoc as either an object or a function. Spreading the result of `mapPropsToOptions` here can happen in both cases - when `options` is an object or a function. In the case of a function though, the return value of the `options` function is already going to be a new object (see https://www.apollographql.com/docs/react/api/react-apollo.html#graphql-query-config-options). So we really only need to spread the result when we're using an `options` object. I'll adjust this a bit.